### PR TITLE
Jayden Paterson - StyleGAN Q9

### PR DESCRIPTION
**Pull request by Paddo0**
Implementation of a StyleGAN model with code for training and displaying generated images based on the OASIS Brain dataset related to project 9.
The StyleGAN pretrained model provided generates an image that indicates some patterns towards emulating the training set, however, further training and adjustment of hyperparameter could improve the results of the StyleGAN. Other improvements which can be made is by testing which application function works best for this specific model, i.e. tanh, sigmoid, and different parameters for LeakyReLU could be tested for further improvements.

### CODE REVIEW

**Comment by WeiDaiwow** [comment]
# Document
 
- Title: 
	Good
- Address problem:
	Problem is not addressed.
- Working Principle:
	Very brief
- Figure visualization
	Good
- dependency
	Not provided
- inputs and outputs description:
	Inputs provided (only one provided)
	Output provided but its quality is not very good.
- training plot:
	Not provided
- readme using markdown:
	Good
- preprocessing description:
	Not provided
- Description of usage:
	Very brief, no details provided.

# Check the files:
- modules.py
	Good
- dataset.py
	Used online OASIS 3D data by loading slices for training. Generator class was not used when loading the images, hence, the number of loaded images was manually limited due to memory limitation. Not all images were used during training. generator class should be used when loading large dataset.
	comments are not very intuitive. 
- train.py
	the training strategy and criterion does match the description in styleGANv1 paper. Normal GAN training procedure was used instead.
	
- predict.py
	Good

Question:
1. Why you permute twice inside the forward function within StyleGANGeneratorBlock 
	


# Overall:

- solve the problem:
	No
- TF/torch implementation:
	Yes
- Design quality:
	The data loader implementation is not correct. The training procedure does not match the paper.
- code comments:
	Comments inside train.py and dataset.py are limited. 
- Difficulty: 
	StgleGANv1 -hard

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, dataloader deficiencies -2
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem, no no evidence of training/validation -2
Driver Script present
File structure not present as need code into folder and not in main trunk of repo -2
Shows Usage & Demo & Visualisation & Data usage, no plots etc. -2
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe minimal, no refs, usage or problem details (deps etc.) -3
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in inCorrect Branch) -2
Feedback required, remove models files and move code into folder with reasonable name -2
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by shakes76** [comment]
Models files are too large and thus this PR wont be merged into main repo. Feedback marks can still be obtained as above.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
